sandbox/apparmor/apparmor_test.go

```diff
@@ -378,3 +393,45 @@ func (s *apparmorSuite) TestValidateFreeFromAAREhappy(c *C) {
 		c.Check(apparmor.ValidateNoAppArmorRegexp(s), IsNil, Commentf("%q raised an error but shouldn't", s))
 	}
 }
+
+func (s *apparmorSuite) TestUpdateHomedirsTunableMkdirFail(c *C) {
+	restore := apparmor.MockMkdirAll(func(string, os.FileMode) error {
+		return errors.New("mkdir failure")
+	})
+	defer restore()
+
+	err := apparmor.UpdateHomedirsTunable([]string{"does", "not", "matter"})
+	c.Check(err, ErrorMatches, `cannot create AppArmor tunable directory: mkdir failure`)
+}
+
+func (s *apparmorSuite) TestUpdateHomedirsTunableWriteFail(c *C) {
```

Comment:
we often don't write this kind of test fwiw, as there's a straight return of the error in the code